packages/react-client/src/ReactFlightClient.js

```diff
@@ -960,6 +980,22 @@ function initializeModelChunk<T>(chunk: ResolvedModelChunk<T>): void {
         return;
       }
     }
+
+    if (__DEV__ && isReactElementOrArrayLike(value)) {
```

Comment:
We talked about the case where the root is lazy but it's immediately resolved because it got unblocked by the cycle resolution. Then we should unwrap it above.

Otherwise this could also be a lazy here and if it was then it should be transferred to the lazy or inside of it.